Add support for generating mag samplesheet (#544)
nf-core/taxprofiler: docs/output.md

Please aware the outputs of each tool's standardised profile _may not_ be directly comparable between each tool. Some may report raw read counts, whereas others may report abundance information. Please always refer to the list above, for which information is used for each tool.
:::

The pipeline can also generate downstream pipeline input samplesheets.

> These are stored in `<outdir>/downstream_samplesheets`.

Reviewer: This should go right at the top in the bullet poiont summary of the tools/directories

Author: Done

</details>

[Nextflow](https://www.nextflow.io/docs/latest/tracing.html) provides excellent functionality for generating various reports relevant to the running and execution of the pipeline. This will allow you to troubleshoot errors with the running of the pipeline, and also provide you with other information such as launch commands, run times and resource usage.

### Downstream samplesheets

> The pipeline can also generate input files for the following downstream
> pipelines:

Reviewer: ```suggestion
The pipeline can also generate input files for the following downstream pipelines:
```

- [nf-core/mag](https://nf-co.re/mag)

<details markdown="1">
<summary>Output files</summary>

- `downstream_samplesheets/`

>   - `mag.csv`: Only for paired-end reads

Reviewer: ```suggestion
  - `mag.csv`: input sheet for nf-core/mag with paths to nf-core/taxprofiler preprocessed (corresponding to what is saved with `--save_analysis_ready_fastqs`)
```

Why not single-end reads too?

>   - `mag.csv`: input sheet for nf-core/mag with paths to nf-core/taxprofiler preprocessed (corresponding to what is saved with `--save_analysis_ready_fastqs`)

Reviewer: ```suggestion
  - `mag.csv`: input sheet for that contains paths to preprocessed FASTQs (corresponding to what is saved with `--save_analysis_ready_fastqs`) that can be used to skip read preprocessing steps in nf-core/mag 
```

>   - `mag.csv`: input sheet for nf-core/mag with paths to nf-core/taxprofiler preprocessed (corresponding to what is saved with `--save_analysis_ready_fastqs`)

Reviewer: Missing differential abundance? @LilyAnderssonLee do you thin kyou could merge your PR into this one at this stage?

<summary>Output files</summary>

- `downstream_samplesheets/`
  - `mag-{pe,se}.csv`: input sheet for single-end and paired-end reads that contains paths to preprocessed short-read FASTQs (corresponding to what is saved with `--save_analysis_ready_fastqs`) that can be used to skip read preprocessing steps in nf-core/mag.

>     - Note: if you merge reads, these will be listed in teh `mag-se.csv`.

Author: Cannot make a suggestion but:

teh `mag-se.csv`. -> the `mag-se.csv`.
